crates/deltalake-core/src/kernel/schema.rs

```diff
@@ -166,6 +166,16 @@ impl StructField {
         self.nullable
     }
 
+    /// Returns the phyiscal name
+    pub fn physical_name(&self) -> &str {
+        let phys_name = self.get_config_value(&ColumnMetadataKey::ColumnMappingPhysicalName);
+        match phys_name {
+            None => &self.name,
+            Some(MetadataValue::String(s)) => s,
+            Some(MetadataValue::Number(_)) => panic!("Unexpected type for physical name"), // Number makes no sense here and is against the spec
```

Comment:
What I meant here is not to panic at all but return a `Result`. 

Assuming a server running this code, if we hit this error in the cofig, we would crash the entire server. But this is an error i would deem recoverable and a server may just want to return a message to the user.

Reply:
I corrected that, I must say I'm still learning a bit about rusty error handling, I hope it's better now.
One thing I noticed is that there are two error enum's, one in the kernel and one outside. I adjusted both, however it's kind of duplicate. Not sure I did everything how it's supposed to be.